### PR TITLE
Hw6 epam lab

### PR DESCRIPTION
**Pull request by b1zman**
Нибейти

### CODE REVIEW

**Review by kislyakdmitry — APPROVED** [review]

File: hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+public class SorterLine {
+
+    public static Map<String, Map<String, Integer>> sort(String line) {
+        String value = line.replaceAll("[^A-Za-zА-Яа-я0-9]", " ");
+        List<String> wordsList = List.of(value.split("\\s+"));
+        Map<String, Map<String, Integer>> wordsMap = new TreeMap<>();
+        for (String word : wordsList) {
+            String firstLetter = word.substring(0, 1).toUpperCase();
+            Map<String, Integer> stringIntegerMap = wordsMap.get(firstLetter);
```
Comment (Jericho-d) [inline hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java]: What is stringIntegerMap? What function does it perform?

File: hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            if (stringIntegerMap != null) {
+                stringIntegerMap.merge(word, 1, Integer::sum);
+            } else {
+                stringIntegerMap = new HashMap<>();
+                stringIntegerMap.put(word, 1);
+                wordsMap.put(firstLetter, stringIntegerMap);
+            }
+        }
+        return wordsMap;
```
Comment (Jericho-d) [inline hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java]: Can you simplify the code? I suggest to split the code into a small methods

File: hw6_epam_lab/src/test/java/com/hashMap/SorterLineTest.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Test
+    void sortMap_whenPutLine() {
+        Map<String, Map<String, Integer>> wordsMap = new TreeMap<>();
+        wordsMap.put("A", Map.of("animal", 2, "At", 1));
+        wordsMap.put("B", Map.of("blood", 3));
+        String value = "At animal blood blood animal blood";
+        assertThat(wordsMap).isEqualTo(SorterLine.sort(value));
+
+    }
```
Comment (Jericho-d) [inline hw6_epam_lab/src/test/java/com/hashMap/SorterLineTest.java:42]: Where are a negative scenarios?

**Review by kislyakdmitry — APPROVED** [review]

File: hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java (hunk 1 above)
Comment (b1zman) [reply hw6_epam_lab/src/main/java/com/hashMap/SorterLine.java]: Go n Done

File: hw6_epam_lab/src/test/java/com/hashMap/SorterLineTest.java (hunk 3 above)
Comment (b1zman) [reply hw6_epam_lab/src/test/java/com/hashMap/SorterLineTest.java:42]: Done

**Pull request merged**